tooltips: Add enhanced emoji tooltips with larger preview.

Ritesh-251 opened the pull request:
### Problem

Emoji tooltips currently use the browser’s native `title` attribute, which:
* Only shows plain text
* Makes fine details in emojis difficult to see
* Is inconsistent with reaction tooltips, which use Tippy

### Solution

Replaced native tooltips with Tippy.js tooltips that:
* Display a larger (48×48px) emoji image
* Show the emoji’s name alongside it
* Use `INSTANT_HOVER_DELAY` for responsive hover behavior

#### Scope implemented in this PR
| UI Area | Status |
|--------|--------|
| Message feed emojis | ✔ Updated |
| Compose preview emojis | ✔ Updated (inherits same handler) |
| User status emojis | ✔ Updated |
| Reaction emojis | ❌ Skipped per issue guidance |

### Implementation details

* Added `emoji_tooltip.hbs` for consistent tooltip UI
* Added CSS styling for enlarged emoji images in tooltip
* Updated `message_list_tooltips.ts` to attach rich tooltips to message and compose preview emojis
* Updated tooltip initialization for status emoji in `tippyjs.ts`
* Removed conflicting default tooltip attributes from status emoji template

### Testing

Manual verification:
* Hovering emojis in message feed shows large emoji + name
* Compose preview emojis show tooltip correctly
* Status emoji hover shows large emoji + name
* Reaction tooltips unchanged

<details>
<summary>📸 Screenshots</summary>

<img width="449" height="155" alt="Screenshot 2025-11-24 at 7 37 39 AM" src="https://github.com/user-attachments/assets/dd5a1733-6c45-4dde-a118-98ad7e487345" />
<img width="362" height="222" alt="Screenshot 2025-11-24 at 7 37 14 AM" src="https://github.com/user-attachments/assets/9b70aba9-ea86-4aaf-a70d-85c9f2a51644" />

</details>

### Self-review checklist

- [x] I have tested all UI changes manually.
- [x] I verified that tooltips show the correct enlarged emoji and name.
- [x] I confirmed compose preview and message feed use the same tooltip handler.
- [x] I confirmed status emoji tooltips initialize correctly via `tippyjs.ts`.
- [x] I ensured reaction tooltips remain unchanged.
- [x] I checked that no `title` attributes remain on updated emoji elements.
- [x] I verified that CSS for enlarged emoji images is scoped and doesn’t affect other UI.
- [x] I reviewed all selectors and ensured they match Zulip’s conventions.
- [x] I ran linters and ensured no unused code or debug statements remain.
- [x] I checked commits for clarity, proper structure, and no fix-up commits.
- [x] I added screenshots for all visual changes.

### Issue

Fixes #35170

Small font size
<img width="1512" alt="Screenshot 2025-12-07 at 4 51 52 AM" src="https://github.com/user-attachments/assets/3ae6f01f-4087-4cfa-93ec-14e52f4f6aea" />
Default font size
<img width="1512" alt="Screenshot 2025-12-07 at 4 51 37 AM" src="https://github.com/user-attachments/assets/fffba8e8-ec7c-47ea-947a-8145ab12de4a" />
Large font size
<img width="1512" alt="Screenshot 2025-12-07 at 4 52 02 AM" src="https://github.com/user-attachments/assets/dcb89ef0-a60f-4476-b754-57c151fdedc6" />

Ritesh-251 committed:
tooltips: Add enhanced emoji tooltips with larger preview.

Implements improved emoji tooltips that show a 48px preview of the
emoji along with its name using Tippy.js with INSTANT_HOVER_DELAY.

- Added emoji_tooltip.hbs template for consistent tooltip rendering
- Implemented tooltips in message feed (excluding reactions)
- Added tooltips for status emojis across all UI locations
- Unicode emojis display actual characters, custom emojis show images
- Added CSS styling for large emoji display with proper font fallbacks

Fixes #35170.

alya commented:
I don't think this is handling different font sizes correctly. I'd expect the display to scale with font size -- it looks too big at small font sizes.

Ritesh-251 committed:
tooltips: Use Tippy for emoji tooltips in message feed and status.

This commit replaces the native browser tooltips (title attribute)
for emojis in the message feed and user status with Tippy.js
tooltips.

This provides a consistent, larger, and more readable tooltip
experience, matching the style of other tooltips in the application.
It also supports displaying the emoji name and a larger preview
of the emoji (either the custom image or the Unicode character).

Fixes #35170.

Ritesh-251 committed:
Improve emoji tooltips and backend markup

Ritesh-251 commented:
Hi @alya 

I’ve completed all requested changes:

✔️ Tooltip sizing now scales with font size instead of using fixed pixels  
✔️ All native `title` attributes have been removed to prevent double tooltips  
✔️ Message feed, compose preview, and status emoji now consistently use Tippy  
✔️ Reaction tooltips remain unchanged as per issue scope  
✔️ All tests and CI checks are passing

This PR should now be ready for final review.  
Could you please remove the `not ready` label when convenient?

Thanks!

Ritesh-251 commented:
Small font size
<img width="1512" alt="Screenshot 2025-12-07 at 4 51 52 AM" src="https://github.com/user-attachments/assets/3ae6f01f-4087-4cfa-93ec-14e52f4f6aea" />
Default font size
<img width="1512" alt="Screenshot 2025-12-07 at 4 51 37 AM" src="https://github.com/user-attachments/assets/fffba8e8-ec7c-47ea-947a-8145ab12de4a" />
Large font size
<img width="1512" alt="Screenshot 2025-12-07 at 4 52 02 AM" src="https://github.com/user-attachments/assets/dcb89ef0-a60f-4476-b754-57c151fdedc6" />

alya commented:
No, closing this PR is counter-productive. Why would you waste my time that way?

Please review the [contributor guide](https://zulip.readthedocs.io/en/latest/contributing/contributing.html) to learn how to pick up issues when you have a PR under review.

karlstolley commented:
@Ritesh-251 please fix the merge conflicts here so I can review

Ritesh-251 commented:
> @Ritesh-251 please fix the merge conflicts here so I can review

@karlstolley done
